mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -47,18 +47,47 @@ object MLTestingUtils extends SparkFunSuite {
     } else {
       genRegressionDFWithNumericLabelCol(spark)
     }
-    val expected = estimator.fit(dfs(DoubleType))
-    val actuals = dfs.keys.filter(_ != DoubleType).map(t => estimator.fit(dfs(t)))
+
+    val expected = estimator match {
+      case weighted: Estimator[M] with HasWeightCol =>
+        weighted.set(weighted.weightCol, "weight")
```

Comment:
You could also move the code dealing with weights outside these 2 sections:
```
finalEstimator = estimator match {
  case weighted: ... =>
    weighted.set(weighted.weightCol, "weight")
  case _ => estimator
}
```
Then the other code could remain unchanged.